utilities/ldap_sync.sql

```diff
@@ -25,118 +152,763 @@ Changes in this version:
+            raise Exception("Either the host " + str(server) + " is not reachable or Port " + str(port) + " is not reachable on " + server, exa.meta.script_name)
+    except socket.error:
+        raise Exception(f"Socket connect returned: {connect_result}.", exa.meta.script_name, (str(socket.error), abortMessage))
+    except Exception as e:
+        raise Exception(f"Results from socket connection on host " + server + " port " + str(port) + " did not work; Return_code {}".format(connect_result), exa.meta.script_name, (str(e), abortMessage))
 
+
+#######################################
+# BEGIN LOGIC
+#######################################
 def run(ctx):
-        # The below information corresponds to the user needed to connect to ldap who can traverse the ldap structure and pull out user attributes. 
-        # This information should be stored in a CONNECTION object and you must GRANT ACCESS ON <CONNECTION> FOR <SCRIPT> TO <USER>
-        # More details: https://docs.exasol.com/database_concepts/udf_scripts/hide_access_keys_passwords.htm
-        uri =   exa.get_connection(ctx.LDAP_CONNECTION).address #ldap/AD server
-	user = exa.get_connection(ctx.LDAP_CONNECTION).user   #technical user for LDAP
-	password = exa.get_connection(ctx.LDAP_CONNECTION).password  #pwd of technical user
-	encoding = "utf8"  #may depend on ldap server, try latin1 or cp1252 if you get problems with special characters
-
-	try:
-		#Sets a timeout of 5 seconds to connect to LDAP
-		ldap.set_option(ldap.OPT_NETWORK_TIMEOUT, 5.0)
-		
-		#The below line is only needed when connecting via ldaps
-		ldap.set_option(ldap.OPT_X_TLS_REQUIRE_CERT, ldap.OPT_X_TLS_NEVER)   # required options for SSL without cert checking
-		
-		# Connects to LDAP
-		ldapClient = ldap.initialize(uri)
-		
-		#Authenticates with user
-		ldapClient.bind_s(user, password)
-	
-		results = ldapClient.search_s(ctx.SEARCH_STRING.encode(encoding), ldap.SCOPE_BASE)
+    #=======================================
+    # House Keeping - Validate LDAP Host Connection 
+    #========================================
+    print("Hello World")
+    
+    if ctx.VERIFY.upper() == 'DEBUG':
+        DEBUG = 1
+    else:
+        DEBUG = 0
+    
+    #----------
+    # Ensure Connection String has a proper LDAP server
+    #----------
+    global uri
+    try:
+        uri = exa.get_connection(ctx.LDAP_CONNECTION).address
+        ldap_server = str(uri)
+    except Exception as e:
+        raise Exception(f"Unable to find/parse {uri} from the exa.get_connection(ctx.LDAP_CONNECTION).address. Error caught in:", exa.meta.script_name, str(e), abortMessage)
+    
+    if DEBUG:
+        validate_connection(ldap_server)
+        
+    
+
+    #========================================
+    # House Keeping - Validate remaining Connection properties
+    #========================================
+    try:
+        user = exa.get_connection(ctx.LDAP_CONNECTION).user        #technical user for LDAP
```

Comment:
Instead of calling exa.get_connection multiple times you can do it once and save the connection object for later usage.